server/src/main/java/org/candlepin/model/ProductCurator.java

```diff
@@ -330,6 +330,17 @@ protected Product validateProductReferences(Product entity) {
             }
         }
 
+        if (entity.getBranding() != null) {
```

Comment:
I wonder if we're getting to a point where we should have an ```isComplete``` method on our entities to check this kind of thing without needing this kind of code in a number of places.